### PR TITLE
Add cryptography, openssl, and _ssl packages

### LINKED ISSUE #761: Cryptography Support
Pyodide is genius. It will certainly influence and broaden the world impact of all these data science tools and Python in so many new contexts. 

Any plan on adding 'cryptography' to the list of supported packages?

It seems many cool modules depend on it for fundamental encryption and security. All applications that would be awesome in the browser with Pyodide. 

e.g.
```console
>>> micropip.install('ccxt')
Couldn't find a pure Python 3 wheel for 'cryptography'
>>> micropip.install('alias')
Couldn't find a pure Python 3 wheel for 'cryptography'
>>> micropip.install('keycache')
Couldn't find a pure Python 3 wheel for 'cryptography'
```

### PR DESCRIPTION
**Pull request by hoodmane**
Resolves #761.

ssl passes 46/168 tests. 108 fail due to inability to start a socket and another 14 failing with attribute errors due to missing library functions.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Review by ryanking13 — COMMENTED** [review]
Thanks, @hoodmane. It's kind of amazing that openssl "just works", though it's obvious that most of the core functionalities are unavailable.

File: packages/ssl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+source:
+  path: empty
+build:
+  sharedlibrary: true
+  script: |
+    mkdir dist
+    export DISTDIR=$(pwd)/dist
+    cd $CPYTHONBUILD
+    emcc $STDLIB_MODULE_CFLAGS -c Modules/socketmodule.c -o Modules/socketmodule.o
+    emcc $STDLIB_MODULE_CFLAGS -c Modules/_ssl.c -o Modules/_ssl.o -I$PYODIDE_ROOT/packages/openssl/build/openssl-1.1.1m/include/ -DOPENSSL_THREADS
```
[the remark this answers is not in the document]
Comment (hoodmane) [reply packages/ssl/meta.yaml:13]: It declares that the openssl implementation is thread safe. With no threads everything is thread safe =)

[the remark this answers is not in the document]
Comment (hoodmane) [reply packages/ssl/meta.yaml:13]: I will add a comment saying this.

**Review by ryanking13 — APPROVED** [review]
LGTM, thanks!

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13.

**Comment by rth** [comment]
Thanks a lot for making this happen !

Could you please just add ssl to https://pyodide.org/en/stable/usage/wasm-constraints.html#optional-modules and maybe we should re-evaluate what to do with `fullStdLib = true`. The current behavior of not loading ssl is probably better but then this flag is a bit confusing as there would be more exceptions that modules that follow it.

**Comment by psytron** [comment]
Wow. Game changing  @hoodmane

**Pull request merged**